src/web/src/views/workspace/WSEditor.tsx

```diff
@@ -674,6 +774,119 @@ class WSEditorSwaggerReloadDialog extends React.Component<WSEditorSwaggerReloadD
 
 }
 
+interface WSRenameDialogProps {
+    workspaceUrl: string,
+    workspaceName: string,
+    open: boolean,
+    onClose: (newWSName: string) => void
+}
+
+interface WSRenameDialogState {
+    newWSName: string,
+    invalidText?: string,
+    updating: boolean
+}
+
+class WSRenameDialog extends React.Component<WSRenameDialogProps, WSRenameDialogState> {
+
+    constructor(props: WSRenameDialogProps) {
+        super(props);
+        this.state = {
+            newWSName: this.props.workspaceName,
+            updating: false
+        }
+    }
+
+    handleModify = (event: any) => {
+        let { newWSName } = this.state;
+        let { workspaceUrl, workspaceName } = this.props;
+
+        let nName = newWSName.trim();
+        if (nName.length < 1){
+            this.setState({
+                invalidText: `Field 'Name' is required.`
+            })
+            return;
+        }
+
+        this.setState({
+            invalidText: undefined
+        });
+        const nodeUrl = `${workspaceUrl}`;
+        this.setState({
+            updating: true,
+        })
+
+        if (workspaceName === nName) {
+            this.setState({
+                updating: false,
+            })
+            this.props.onClose("");
+        } else {
+            axios.post(`${nodeUrl}/Rename`, {
+                name: nName
+            }).then(res => {
+                console.log(res);
+                this.setState({
+                    updating: false,
+                })
+                this.props.onClose(res.data.name);
+            }).catch(err => {
+                console.error(err.response.data);
+                this.props.onClose("");
+            })
+        }
+        
+    }
+
+    handleClose = () => {
+        this.setState({
+            invalidText: undefined
+        });
+        this.props.onClose("");
```

Comment:
recommend using null
```suggestion
        this.props.onClose(null);
```